src/imageview.cpp

```diff
@@ -307,8 +318,11 @@ void ImageView::setImage(const QImage& image, bool show) {
     zoomFit();
   queueGenerateCache();
 
-  // clear the numbering
+  // clear the numbering and annotation items in the scene
   nextNumber = 1;
+  for (QGraphicsItem *annotation : annotations){
+      scene_->removeItem(annotation);
+  }
```

Comment:
There are three problems here:

 1. This can cause crash with SVG and GIF because `scene_->clear();` is used for them near the beginning of the current function, which makes each annotation a dangling pointer.
 2. There is a memory leak because `QGraphicsScene::removeItem()` doesn't delete the item.
 3. `annotations` should be cleared here.

So, I suggest you add `annotations.clear();` after `scene_->clear();` (see above) and use these lines here:
```c++
  // clear the numbering and annotation items in the scene
  nextNumber = 1;
  if(!annotations.isEmpty()) {
    for(QGraphicsItem *annotation : qAsConst(annotations)) {
        scene_->removeItem(annotation);
    }
    qDeleteAll(annotations.begin(), annotations.end());
    annotations.clear();
  }
```

Reply:
@tsujan you're right, it will cause crash and memory leak
 I have fixed it. The new code is almost like the code you suggest, but with a minor adjustment. 
because we can also annotate gif image, the annotations list may also not be empty when it's gif image, so i put the code to the beginning of the function. as the commit log describes:

>  In setImage() function, when the image is gif or svg, scene_->clear()
    is called, which clear and delete the annotation items, so we just
    need clear the annotations list, delete or remove the annotations
    will make a crash
    on the other hand, for a normal image, scene_->removeItem() just
    remove the item but not delete it, so we need delete it by ourself
    to prevent a memory leak
    At last, we need clear the annotations list for both situations

Reply:
> At last, we need clear the annotations list for both situations

That's what I suggested above.

So far so good. I didn't find another problem by a glance at the PR. Will read its details soon.